# SPIDER-seq {.unnumbered}

Single-cell Projectome-transcriptome In situ Deciphering Sequencing

![movie](images/movie.gif)



## Summary

Deciphering the connectivity, transcriptome, and spatial-omics integrated multi-module brain atlas and the underlying organization principles remains challenging. We developed a cost-effective **Single-cell Projectome-transcriptome In situ Deciphering Sequencing (SPIDER-Seq)** method by combining viral barcoding tracing with single-cell sequencing and spatial-omics. Leveraging SPIDER-Seq, we delineated a comprehensive integrated single-cell spatial molecular, cellular and projectomic atlas of mouse prefrontal cortex (PFC). The projectomic and transcriptomic cell clusters display distinct organization principle, but are coordinately configured in PFC. These projection neurons gradiently distributed in PFC aligning with their wiring patterns, and importantly, show higher co-projection probability to the downstream nuclei with reciprocal circuit connections. Moreover, we depicted PFC neural transmission map, in which neural transmission molecues (neurotransmitter/neuropeptide and the receptors) distinctly express in different circuits. Finally, we predicted neuron projections by integrated gene profile and spatial information with high accuracy via machine learning. Our study could greatly facilitate to delineating brain multi-module network and understanding neural computation.

![movie](images/Abstract.jpg)



## Data Visualization

We developed is a shiny application **SPIDER-web** that allows users to interactively access our data. You can access our online website at <https://huggingface.co/spaces/TigerZheng/SPIDER-web>



## Code View

The code of this project can be viewed at <https://zhengtiger.github.io/SPIDER-seq>



## Citation

Our project has been released in the following publication:

